nav2_navfn_planner/include/nav2_navfn_planner/navfn_planner.hpp

```diff
@@ -29,18 +29,19 @@
 #include "nav2_msgs/msg/path.hpp"
 #include "nav2_navfn_planner/navfn.hpp"
 #include "nav2_util/costmap_service_client.hpp"
-#include "nav2_util/get_robot_pose_client.hpp"
+#include "nav2_util/robot_utils.hpp"
 #include "nav2_util/simple_action_server.hpp"
 #include "nav_msgs/msg/path.hpp"
 #include "visualization_msgs/msg/marker.hpp"
+#include "tf2_ros/transform_listener.h"
 
 namespace nav2_navfn_planner
 {
 
 class NavfnPlanner : public nav2_util::LifecycleNode
 {
 public:
-  NavfnPlanner();
+  NavfnPlanner(std::shared_ptr<tf2_ros::Buffer> tf);
```

Comment:
Right now, we're not really using any type of plugins for the planners / controllers / recoveries but I figure at some point in the near future we'll need to standardize and API for running those processes and exposing consistent servers. When we hit that point a TF buffer should be something given to the constructor of that plugin so that a process loading many "things" can share a single buffer.

I do agree in this case for how the code is its a little funny. If it bothers you, I can make the buffer in the NavFnPlanner constructor as well, however in a PR impending I'm moving the global costmap into the navfn process so sharing that buffer would be best, but I could also rethink how I'm doing that.

Reply:
It is a little funny for now, though I agree with you on the plugin model. For now I think it's cleaner to have it create its own buffer given the current model. 

On NavFn and costmap sharing buffer in same process, I think that should be opened as an issue. Right now we're using the `WorldModel` to host the global costmap and distribute across ROS communication. I believe we purposely moved toward this model in consideration of a more complete world model (isolating from those who are clients of it), though I concede there may be drawbacks as well, so we should reopen the discussion if we're going to move in the other direction.

Reply:
Ok, I'll move the buffer in the object, that's fine